provider/consumer.py

```diff
@@ -210,73 +198,66 @@ def run(self):
         except Exception as e:
             logging.error('[{}] Uncaught exception: {}'.format(self.trigger, e))
         finally:
+            logging.info('[{}] Cleaning up consumer'.format(self.trigger))
+            logging.debug('[{}] Closing KafkaConsumer'.format(self.trigger))
+            self.consumer.unsubscribe()
```

Comment:
@dubeejw This is addressed in c024c451827c69d1ffb54459c1ac67ee5cab5db6. Because the previous try/catch catches all exceptions, the cleanup code can be moved outside of the `finally` block and it will behave the same. This allows any possible exceptions to be caught, without the need for nested try/catch blocks.